src/inmanta/ast/type.py

```diff
@@ -436,6 +436,9 @@ def validate(self, value: Optional[object]) -> bool:
 
         return True
 
+    def type_string(self) -> str:
+        return "list"
```

Comment:
This slightly changes the semantics of this class. I don't like that we now have two distinct classes returning `"list"` here, but that's more a property of the language (`list` in DSL vs `list` in plugin annotation) than anything else, so I think you took the right approach. That said, could you extend the docstring with something like "e.g. the `list` type in plugin annotations."? And perhaps refer to `LiteralList` for the DSL `list` type?

Also, an additional comment here could help future confusion.

```suggestion
        # this is not a type in the model, but it is used in plugin annotations, which are also part of the DSL.
        return "list"
```